Update help me choose to include ADX (#1097)
microsoft/finops-toolkit: docs/_reporting/power-bi/README.md

When you need more advanced reporting or to merge with your own data, we recommend using Microsoft Fabric, Power BI, or a custom or third-party solution. Use the following to determine the best approach for you:

> 1. For costs under $2-5M in total<sup>1</sup> that don't need savings plan data, you can use the Cost Management connector for Power BI.

Reviewer: I think we need to keep this. It's still a reality and default expectation for many customers. We need to convey why it's not a good idea.

   - The connector uses existing raw cost data APIs and cannot scale to data sizes beyond $5M<sup>1</sup>.

>    - Due to the size constraints, the connector will be phased out by the Cost Management team starting in 2024.

Reviewer: We should change this to 2025

> 1. **Power BI on Cost Exports**: For costs under $2-5M/month (~$65M total)<sup>2</sup> that need savings plan data, you can use raw exports with Power BI.

Reviewer: ```suggestion
1. **Power BI on Cost Management exports**: For costs under $2-5M/month (~$65M total)<sup>2</sup> that need savings plan data, you can use raw exports with Power BI.
```

> 2. **FinOps Hubs with Azure Data Factory**: For costs under $2-5M/month (~$65M total)<sup>2</sup> that need savings plan data, you can use FinOps hubs with Power BI.

Reviewer: ```suggestion
2. **FinOps hubs on storage**: For costs under $2-5M/month (~$65M total)<sup>2</sup> that require support for multiple Entra ID tenants or prefer the solution to manage exports for you, you can use FinOps hubs with Power BI.
```

> 3. 🆕 **FinOps Hubs with Azure Data Explorer (ADX)**:  For scenarios where you need more advanced exploration, Hubs on ADX allows you to leverage Kusto Query Language (KQL) for in-depth analysis and provides built-in queries to simplify data exploration. ADX also offers scalability, allowing you to increase capacity by adjusting the virtual machine size. The default VM size supports up to $10M/month of cost data.

Reviewer: I don't think we need to get into sizing here. We just need to convey why they should choose this option over one before it.
```suggestion
3. 🆕 **FinOps hubs with Azure Data Explorer**:  For costs over $2-5M/month or when you need more advanced exploration, we recommend FinOps hubs with Azure Data Explorer (ADX). Leverage Kusto Query Language (KQL) for advanced analytics and improved performance and scalability.
```

4. _**Coming soon:**_ For costs over $5M/month or for additional capabilities, you can connect Fabric to either FinOps hubs or raw exports.
   - This is possible today, but is not supported in FinOps toolkit reports yet. Support will be added in a future release.

_<sup>1) Power BI Pro can handle ~$2M of raw cost data. Power BI Premium can handle ~$5M.</sup>_

_<sup>2) The $2-5M limits are for Power BI data refreshes and apply on a monthly basis for hubs and raw exports. They can load up to $65M with incremental refresh enabled.</sup>_

> _<sup>3) ❗ The Cost Management connector is no longer recommended as it will be deprecated soon.</sup>_

Reviewer: This isn't referenced anywhere. Not sure we need it if we're explicitly calling out what we do recommend. If we use something like this, we should keep it with the section above.
```suggestion
```

_<sup>1) Power BI Pro can handle ~$2M of raw cost data. Power BI Premium can handle ~$5M.</sup>_

_<sup>2) The $2-5M limits are for Power BI data refreshes and apply on a monthly basis for hubs and raw exports. They can load up to $65M with incremental refresh enabled.</sup>_

In general, we recommend starting with the Cost Management connector when getting started with Power BI reports. The most common reasons to switch to FinOps hubs are for additional account types and scopes or to enable more advanced capabilities. Use the following comparison to help you make the decision:
_<sup>3) ❗ The Cost Management connector is no longer recommended as it will be deprecated soon.</sup>_

> | Capabilities                                        |            Connector             |             Exports              |           FinOps hubs            | Microsoft Fabric<sup>1</sup> |

Reviewer: After getting thru the table, I realized you got rid of the connector. We need to keep that.

> In general, we recommend starting with Power BI reports by connecting to the cost exports. The most common reasons to switch to FinOps hubs are for additional account types and scopes or to enable more advanced capabilities. Use the following comparison to help you make the decision:

Reviewer: ```suggestion
In general, we recommend starting with Power BI reports by connecting to the Cost Management exports. The most common reasons to switch to FinOps hubs are for performance, scale, and to enable more advanced capabilities. Use the following comparison to help you make the decision:
```

> | Capabilities                                        |             Power BI on Cost Exports              |        FinOps hubs with ADF         |        FinOps hubs with ADF and ADX         | Microsoft Fabric<sup>1</sup> |

Reviewer: ```suggestion
| Capabilities                                        |             Exports              |        FinOps hubs (storage)         |        FinOps hubs (ADX)         | Microsoft Fabric<sup>1</sup> |
```

> | Capabilities                                        |             Power BI on Cost Exports              |        FinOps hubs with ADF         |        FinOps hubs with ADF and ADX         | Microsoft Fabric<sup>1</sup> |

Reviewer: The further down this list I get, the more I wonder if we should even call out hubs on storage 🤔

> | Cost (based on list prices)                         |           ~$10 per $1M           |           ~$25 per $1M           |      ~$220 per month<sup>2</sup>  |             TBD              |

Reviewer: Let's use the dev/test pricing, since that's default and all most people will need.
```suggestion
| Monthly cost (based on list prices)                 |           ~$23 per $1M           |           ~$25 per $1M           |      $120 + ~$30 per $1M<sup>2</sup>  |             $300+              |
```

> | Est. max raw cost details per month<sup>3</sup>     | $2M/mo (Pro)<br>$5M/mo (Premium) | $2M/mo (Pro)<br>$5M/mo (Premium) |                                  |             TBD              |

Reviewer: ```suggestion
| Est. max raw cost details per month<sup>3</sup>     | $2M/mo (Pro)<br>$5M/mo (Premium) | $2M/mo (Pro)<br>$5M/mo (Premium) |              TBD                 |             TBD              |
```

> | Est. max total with incremental refresh<sup>4</sup> |   $2M (Pro)<br>$65M (Premium)    |   $2M (Pro)<br>$65M (Premium)    |                                  |             TBD              |

Reviewer: ```suggestion
| Est. max total with incremental refresh<sup>4</sup> |   $2M (Pro)<br>$65M (Premium)    |   $2M (Pro)<br>$65M (Premium)    |              TBD                 |             TBD              |
```

| Microsoft Partner Agreement                         |                ✅                 |                ✅                 |             ✅                     |              ✅               |
| Microsoft Online Services Agreement                 |                ❌                 |                ❌                 |             ❌                     |              ❌               |
| Billing accounts                                    |                ✅                 |                ✅                 |             ✅                     |              ✅               |
| Billing profiles                                    |                ✅                 |                ✅                 |             ✅                     |              ✅               |
| Invoice sections                                    |                ✅                 |                ✅                 |             ✅                     |              ✅               |
| CSP customers (partner only)                        |                ✅                 |                ✅                 |             ✅                     |              ✅               |

> | Management groups                                   |                ❌                 |                ❌                 |             ✅                     |              ❌               |

Reviewer: ```suggestion
| Management groups                                   |                ❌                 |                ❌                 |             ❌                     |              ❌               |
```

| Subscriptions                                       |                ✅                 |                ✅                 |             ✅                     |              ✅               |
| Resource groups                                     |                ✅                 |                ✅                 |             ✅                     |              ✅               |

> | Supports savings plans<sup>4</sup>                  |                ✅                 |                ✅                 |             ✅                     |              ✅               |

Reviewer: ```suggestion
| Calculate EA and MCA cost savings                  |                ❌                 |                ❌                 |             ✅                     |              ✅ (via Hubs with ADX)               |
| Supports savings plans<sup>4</sup>                  |                ✅                 |                ✅                 |             ✅                     |              ✅               |
```

> | Supports savings plan recommendations               |                ❌                 |                🔜                 |             ✅                     |              🔜               |

Reviewer: ```suggestion
| Supports savings plan recommendations               |                ❌                 |                🔜                 |             🔜                     |              🔜               |
```

| Supports multiple scopes                            |                ✅                 |                ✅                 |             ✅                     |              ✅               |

> | Supports scopes in different tenants<sup>5</sup>    |          ⚠️<sup>5</sup>           |          ⚠️<sup>5</sup>           |             ❌                     |          ⚠️<sup>5</sup>(via Hubs) |

Reviewer: ```suggestion
| Supports scopes in different tenants<sup>5</sup>    |          ❌<sup>5</sup>           |          ✅           |             ✅                     |          ✅ (via Hubs) |
```

> | Faster data load times                              |                🔜                 |                ✅                 |             ✅                     |         ✅ (via Hubs)         |

Reviewer: ```suggestion
| Faster data load times                              |                ❌                 |                ❌                 |             ✅                     |         ✅         |
```

> | Supports >$65M in cost details                      |                ❌                 |                ✅                 |             ❌                     |              ✅               |

Reviewer: ```suggestion
| Supports >$65M in cost details                      |                ❌                 |                ❌                 |             ✅                     |              ✅               |
```

| Accessible outside of Power BI                      |                ✅                 |                ✅                 |             ✅                     |              ✅               |

> | Kusto Query Language (KQL) support                        |                ❌                 |                ❌                 |             ✅                     |              ❌               |

Reviewer: ```suggestion
| Kusto Query Language (KQL) support                        |                ❌                 |                ❌                 |             ✅                     |              ✅               |
```

> | Native integration with cost optimization workbook |         ❌                 |                ❌                 |             ✅<sup>6</sup>                     |              ❌               |

Reviewer: Should we expand this to workbooks in general? Then that's possible today. It's just not something we offer. Beyond that, I'm not sure the COW is big enough to really call out as a feature someone's going to use to make a decision 🤔
```suggestion
| Native integration with cost optimization workbook |         ❌                 |                ❌                 |             🔜                     |              ❌               |
```

> | Azure Data Explorer built-in dashboards                    |                ❌                 |                ❌                 |             ✅                     |              ❌               |

Reviewer: This doesn't feel applicable in a Power BI doc.
```suggestion
```

Author: THat is why I suggested this doc to be somewhere else. This is a help me choose section for Hubs.

Reviewer: But this isn't about hubs. Right now it's about the options available for using Power BI. If we want to expand it to include other reporting options, then those would need to be columns and not rows. I would remove this for now since we don't even have any dashboards available out of the box.

| Learn more                                          |      [Learn more][about-rawexports]    |     [Learn more][about-hubs]     | [Learn more][about-hubs]          | [Learn more][about-hubs]       |

[about-connector]: https://aka.ms/costmgmt/powerbi
[about-hubs]: ../hubs/README.md
[about-rawexports]: ../power-bi/setup.md
[hubs-template]: ../hubs/template.md

_<sup>1) Microsoft Fabric can connect to either raw exports or FinOps hubs. FinOps toolkit reports do not support Microsoft Fabric yet but will in a future release.</sup>_

> <sup>2) The estimated cost of Hubs on ADX assumes the two instances are running 24x7. Costs can be reduced by enabling auto-shutdown outside of business hours or applying commitment discounts. Prices are estiamted in UK South as of October 2024</sup>_

Reviewer: Can you reword this to say that the estimate assumes a single node dev/test cluster and that they may need to consider additional nodes for larger accounts? I don't know what the precise guidance should be, so we'll have to be hand-wavy.

| Learn more                                          |      [Learn more][about-rawexports]    |     [Learn more][about-hubs]     | [Learn more][about-hubs]          | [Learn more][about-hubs]       |

[about-connector]: https://aka.ms/costmgmt/powerbi
[about-hubs]: ../hubs/README.md
[about-rawexports]: ../power-bi/setup.md
[hubs-template]: ../hubs/template.md

_<sup>1) Microsoft Fabric can connect to either raw exports or FinOps hubs. FinOps toolkit reports do not support Microsoft Fabric yet but will in a future release.</sup>_

_<sup>2) Power BI constraints are based on data size and processing time. Monitored spend estimations are for reference only. You may see different limits based on services you use and other datasets you ingest.</sup>_
<sup>2) The estimated cost of Hubs on ADX assumes the two instances are running 24x7. Costs can be reduced by enabling auto-shutdown outside of business hours or applying commitment discounts. Prices are estiamted in UK South as of October 2024</sup>_

> _<sup>3) The Cost Management connector for Power BI does not support incremental refresh, so the limits are the same as the per-month estimation. The FinOps hub estimate is based on incremental refresh being enabled, which requires additional configuration after your report is published.</sup>_

Reviewer: Bring this back.

> _<sup>3) Power BI constraints are based on data size and processing time. Monitored spend estimations are for reference only. You may see different limits based on services you use and other datasets you ingest.</sup>_

Reviewer: I like this note, but is it referenced above? It might belong somewhere else.

_<sup>3) Power BI constraints are based on data size and processing time. Monitored spend estimations are for reference only. You may see different limits based on services you use and other datasets you ingest.</sup>_

> _<sup>4) The Cost Management connector uses an old API version and does not include details for some features, like savings plans. Please use FinOps hubs for the latest version with all details.</sup>_

Reviewer: Bring this back.

> _<sup>4) The FinOps hub estimate is based on incremental refresh being enabled, which requires additional configuration after your report is published.</sup>_

Reviewer: Where is this referenced? I'm not sure it's needed since we have a reference to incremental refresh earlier.

> _<sup>6) _Comming soon_. Dashboards will be available in a future release.</sup>_
> 

Reviewer: Not needed. The icon conveys this already.
```suggestion
```

> If you're unsure where to start, we recommend downloading the Power BI dashboards and connecting them to cost exports. This will allow you to explore the reports and see how they work with your data. Alternatively, you can open the Power BI reports using the provided sample data.

Reviewer: ```suggestion
If you're unsure where to start, we recommend downloading the Power BI dashboards and connecting them to Cost Management exports in storage. This will allow you to explore the reports and see how they work with your data. Alternatively, you can open the Power BI reports using the provided sample data.

```

> Our recommendation is to use Hubs on ADX, as it offers exclusive features not available in other versions. These include querying cost data with KQL and native integration with the Cost Optimization workbook. However, if you don't require any of ADX's exclusive capabilities, you can opt for Hubs on ADF instead.

Reviewer: Not sure we need to say some of this again.
```suggestion
For the best performance and capabilities, we recommend using FinOps hubs with Data Explorer, as it offers exclusive features not available in other options listed above.
```
